### DISCUSSION TITLE
Export csv file based on roles and customise csv format

### DISCUSSION THREAD

**Discussion by allenmycho0223** [directus/directus#7645] [Archive]
Hi,

I was wondering if it's possible to customise the csv file format when exporting collections. The reason being that we store timestamp as unix timestamp (bigint) but we display the timestamp as DateTime string on the front end as seen.
![image](https://user-images.githubusercontent.com/88812641/131079038-b7210add-5e05-48ff-b9b0-af2c8c45add1.png)
![image](https://user-images.githubusercontent.com/88812641/131079571-f885fbf7-cf60-45ef-aa03-076c7ac2880b.png)

Hence, after exporting the collections, the timestamps are displayed as bigint which doesn't look user-friendly. Can we customise the csv format before exporting it?

Also, I'd like to limit the export feature as per roles. We have 4 roles including DEVAPPS, Wizard, Admin, and Support. I'd like only DEVAPPS and Wizard users to be able to export collections. Is it also possible?

Any answers would be much appreciated. Thanks in advance!

**Comment by rijkvanzanten** [comment]
Heya! The export functionality in the API simply exports the current request in the given data format (eg csv). Therefore, there's no data transformations (yet), nor dedicated permissions for exporting explicitly (seeing it's just saving the output for a regular `read`)
